src/main/java/net/sf/mzmine/modules/peaklistmethods/alignment/adap3/ADAP3AlignerTask.java

```diff
@@ -0,0 +1,419 @@
+ *
+ * You should have received a copy of the GNU General Public License along with MZmine 2; if not,
+ * write to the Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301
+ * USA
+ */
+package net.sf.mzmine.modules.peaklistmethods.alignment.adap3;
+
+import dulab.adap.common.algorithms.machineleanring.OptimizationParameters;
+import dulab.adap.datamodel.Component;
+import dulab.adap.datamodel.Peak;
+import dulab.adap.datamodel.PeakInfo;
+import dulab.adap.datamodel.Project;
+import dulab.adap.datamodel.ReferenceComponent;
+import dulab.adap.datamodel.Sample;
+import dulab.adap.workflow.AlignmentParameters;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import net.sf.mzmine.datamodel.DataPoint;
+import net.sf.mzmine.datamodel.Feature;
+import net.sf.mzmine.datamodel.IsotopePattern;
+import net.sf.mzmine.datamodel.MZmineProject;
+import net.sf.mzmine.datamodel.PeakIdentity;
+import net.sf.mzmine.datamodel.PeakList;
+import net.sf.mzmine.datamodel.PeakListRow;
+import net.sf.mzmine.datamodel.RawDataFile;
+import net.sf.mzmine.datamodel.impl.SimplePeakInformation;
+import net.sf.mzmine.datamodel.impl.SimplePeakList;
+import net.sf.mzmine.datamodel.impl.SimplePeakListRow;
+import net.sf.mzmine.modules.peaklistmethods.qualityparameters.QualityParameters;
+import net.sf.mzmine.parameters.ParameterSet;
+import net.sf.mzmine.taskcontrol.AbstractTask;
+import net.sf.mzmine.taskcontrol.TaskStatus;
+
+/**
+ * @author aleksandrsmirnov
+ */
+public class ADAP3AlignerTask extends AbstractTask {
+
+    private static final Logger LOG = Logger.getLogger(
+            ADAP3AlignerTask.class.getName());
+
+    private final MZmineProject project;
+    private final ParameterSet parameters;
+
+    private final PeakList[] peakLists;
+
+    private final String peakListName;
+
+    private final Project alignment;
+
+    public ADAP3AlignerTask(MZmineProject project, ParameterSet parameters) {
+
+        this.project = project;
+        this.parameters = parameters;
+
+        this.peakLists = parameters.getParameter(
+                ADAP3AlignerParameters.PEAK_LISTS)
+                .getValue().getMatchingPeakLists();
+
+        this.peakListName = parameters.getParameter(
+                ADAP3AlignerParameters.NEW_PEAK_LIST_NAME).getValue();
+
+        this.alignment = new Project();
+    }
+
+    @Override
+    public String getTaskDescription() {
+        return "ADAP Aligner, " + peakListName + " (" + peakLists.length
+                + " peak lists)";
+    }
+
+    @Override
+    public double getFinishedPercentage() {
+        return alignment.getProcessedPercent();
+    }
+
+    @Override
+    public void run() {
+
+        if (!isCanceled()) {
```

Comment:
Could you change these long blocks of 
```if (!isCanceled()) { ... ```
to much shorter statements of 
```
if (isCanceled()) return;
```
I mean, the function is the same, but the second one makes it much easier to follow the code formatting.